ports/unix/unix_mphal.c

```diff
@@ -220,3 +220,14 @@ uint64_t mp_hal_time_ns(void) {
     gettimeofday(&tv, NULL);
     return (uint64_t)tv.tv_sec * 1000000000ULL + (uint64_t)tv.tv_usec * 1000ULL;
 }
+
+void mp_hal_delay_ms(mp_uint_t ms) {
+    mp_uint_t start = mp_hal_ticks_ms();
+    while (mp_hal_ticks_ms() - start < ms) {
+        #ifdef MICROPY_EVENT_POLL_HOOK
+        MICROPY_EVENT_POLL_HOOK
+        #endif
+        // Sleep for 1ms.
+        usleep(1000);
```

Comment:
since MICROPY_EVENT_POLL_HOOK sleeps for 0.5ms perhaps this can be:
```c
#ifdef MICROPY_EVENT_POLL_HOOK
MICROPY_EVENT_POLL_HOOK
#else 
usleep(1000);
#endif 
```

Reply:
Done